arangodb-net-standard.Test/CursorApi/CursorApiClientTest.cs

```diff
@@ -159,6 +160,18 @@ public async Task PostCursorAsync_ShouldThrow_WhenAqlIsNotValid()
             Assert.Equal(1203, ex.ApiError.ErrorNum);
         }
 
+        [Fact]
+        public async Task PostCursorAsync_ShouldThrowException_WhenDeserializationFailed()
```

Comment:
Ideally we need 3 tests, though I am not sure how to trigger serialization errors for requests yet 🤔 
For the response, it's easier to do because we can mock the json entirely.

Can you rename the current method to `PostCursorAsync_ShouldThrowException_WhenResponseDeserializationFailed` and add the following test:

```
[Fact]
public async Task PostCursorAsync_ShouldThrow_WhenErrorDeserializationFailed()
{
    var mockTransport = new Mock<IApiClientTransport>();

    var mockResponse = new Mock<IApiClientResponse>();

    var mockResponseContent = new Mock<IApiClientResponseContent>();

    string mockJsonError = "{ errorNum: \"some_error\" }";

    mockResponseContent.Setup(x => x.ReadAsStreamAsync())
        .Returns(Task.FromResult<Stream>(
            new MemoryStream(Encoding.UTF8.GetBytes(mockJsonError))));

    mockResponse.Setup(x => x.Content)
        .Returns(mockResponseContent.Object);

    mockResponse.Setup(x => x.IsSuccessStatusCode)
        .Returns(false);

    mockTransport.Setup(x => x.PostAsync(
        It.IsAny<string>(),
        It.IsAny<byte[]>()))
        .Returns(Task.FromResult(mockResponse.Object));

    var cursorApi = new CursorApiClient(mockTransport.Object);

    var ex = await Assert.ThrowsAsync<SerializationException>(async () =>
    {
        await cursorApi.PostCursorAsync<object>("RETURN true");
    });

    Assert.NotNull(ex.Message);
    Assert.NotNull(ex.InnerException);
}
```

Reply:
I already had a test for Deserialization, without the any mocking. I did however change the name as request.

Reply:
Thanks. The test you added only cover deserialization of the response in case of success. It does not cover the deserialization of the error response.

Note in your changes that three methods are affected. The test I posted covers the updated `GetApiErrorException` method, so I think we should add it.

Reply:
right, got it.